bots/goodbot.py

```diff
@@ -41,163 +41,170 @@ def process(self, msg):
 
 		print(str(content) + "\nSuccessfully heard.")
 
-		if sender_email == "notification-bot@zulip.com" and topic == "signups":  # hack for reading #announce stream
-			username = re.match(r"\*\*\.*\*\*", msg["content"]).group(0)
-			self.client.send_message({
-				"type": message_type,
-				"topic": topic,
-				"to": destination,
-				"content": f"Hey @{username} ! Welcome to Wikimedia Zulipchat.\nIf you need any help with GSoD proposals, type `!help gsod`.\nIf you need any help with GSoC proposals, type `!help gsoc`.\nIf you need help with Outreachy proposals, type `!help outreachy`.\nYou can join the #**technical-support** channel to get technical help related to Wikimedia infrastructure.\nType `!help` for a full list of available commands. :blush:"
-			})
-
-		if content[0].lower() == "!help" or content[0] == "@**goodbot**":
-			if(len(content) == 1):
+		try:
+			if sender_email == "notification-bot@zulip.com" and topic == "signups":  # hack for reading #announce stream
+				username = re.search(r"\*\*.*\*\*", msg["content"])
+				if username is None:
+					return
+				else:
+					username = "@" + username.group(0)
 				self.client.send_message({
 					"type": message_type,
 					"topic": topic,
 					"to": destination,
-					"content": f"Hey there, @**{sender_full_name}** ! :blush: Here's what I can do for you:\nType `!help gsod` for help with GSoD proposals.\nType `!help gsoc` for help with GSoC proposals.\nType `!help outreachy` for help with Outreachy proposals.\nType `!help faq 'your question'` to search FAQs.\nType `!help wikipedia 'title'` to search articles.\nType `!help stackoverflow 'your question'` to search questions.\nType `!help chat mediawiki` to get help regarding MediaWiki software.\nType `!help chat wikimedia` to get technical help regarding Wikimedia. You can also join the #**technical-support** channel to get technical help related to Wikimedia infrastructure."
+					"content": f"Hey {username} ! Welcome to Wikimedia Zulipchat.\nIf you need any help with GSoD proposals, type `!help gsod`.\nIf you need any help with GSoC proposals, type `!help gsoc`.\nIf you need help with Outreachy proposals, type `!help outreachy`.\nYou can join the #**technical-support** channel to get technical help related to Wikimedia infrastructure.\nType `!help` for a full list of available commands. :blush:"
```

Comment:
Let's see if there is a way for us to read all these messages from a single file, something like a .json file or something.

Reply:
On it!